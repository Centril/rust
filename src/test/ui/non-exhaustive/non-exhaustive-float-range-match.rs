#![allow(illegal_floating_point_literal_pattern)]
#![deny(unreachable_patterns)]

fn main() {
    match 0.0 {
      0.0..=1.0 => {}
      //~^ WARN floating-point types cannot be used in patterns
      //~| WARN this was previously accepted
      //~| WARN hard error
      //~| WARN floating-point types cannot be used in patterns
      //~| WARN this was previously accepted
      //~| WARN hard error
      _ => {} // ok
    }

    match 0.0 { //~ ERROR non-exhaustive patterns
      0.0..=1.0 => {}
      //~^ WARN floating-point types cannot be used in patterns
      //~| WARN this was previously accepted
      //~| WARN hard error
      //~| WARN floating-point types cannot be used in patterns
      //~| WARN this was previously accepted
      //~| WARN hard error
    }
}
