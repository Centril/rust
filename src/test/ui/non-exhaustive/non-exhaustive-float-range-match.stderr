warning: floating-point types cannot be used in patterns
  --> $DIR/non-exhaustive-float-range-match.rs:6:7
   |
LL |       0.0..=1.0 => {}
   |       ^^^
   |
note: lint level defined here
  --> $DIR/non-exhaustive-float-range-match.rs:1:10
   |
LL | #![allow(illegal_floating_point_literal_pattern)]
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: #[warn(illegal_floating_point_literal_pattern)] is the minimum lint level
   = note: the lint level cannot be reduced to `allow`
   = warning: this was previously accepted by the compiler but is being phased out
   = warning: it will become a hard error in a future release!
   = note: for more information, see issue #41620 <https://github.com/rust-lang/rust/issues/41620>

warning: floating-point types cannot be used in patterns
  --> $DIR/non-exhaustive-float-range-match.rs:6:13
   |
LL |       0.0..=1.0 => {}
   |             ^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out
   = warning: it will become a hard error in a future release!
   = note: for more information, see issue #41620 <https://github.com/rust-lang/rust/issues/41620>

warning: floating-point types cannot be used in patterns
  --> $DIR/non-exhaustive-float-range-match.rs:17:7
   |
LL |       0.0..=1.0 => {}
   |       ^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out
   = warning: it will become a hard error in a future release!
   = note: for more information, see issue #41620 <https://github.com/rust-lang/rust/issues/41620>

warning: floating-point types cannot be used in patterns
  --> $DIR/non-exhaustive-float-range-match.rs:17:13
   |
LL |       0.0..=1.0 => {}
   |             ^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out
   = warning: it will become a hard error in a future release!
   = note: for more information, see issue #41620 <https://github.com/rust-lang/rust/issues/41620>

error[E0004]: non-exhaustive patterns: `_` not covered
  --> $DIR/non-exhaustive-float-range-match.rs:16:11
   |
LL |     match 0.0 {
   |           ^^^ pattern `_` not covered
   |
   = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms

error: aborting due to previous error

For more information about this error, try `rustc --explain E0004`.
